src/DocumentFormat.OpenXml/Framework/Schema/CompiledParticle.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using DocumentFormat.OpenXml.Framework.Schema;
+using DocumentFormat.OpenXml.Validation.Schema;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace DocumentFormat.OpenXml.Framework
+{
+    internal class CompiledParticle : IComparer<OpenXmlElement>
+    {
+        private LookupItem[] _lookup;
```

Comment:
Would a `Dictionary<Type, string>` be better? As it is, `GetPath` operations are O(n) now instead of being indexing operations like they were previously. The collections are small so it may not really matter, but a dictionary might fit the bill (especially since I *think* there can only be one path per type).

Reply:
Since the majority of these are only a few items, my experience in this code base has been to default to lists and do the `O(n)` lookup even though some may be larger just for the overhead that a dictionary imposes. On a similar code path (the one for attributes), I got something like 20% decrease in allocations and speed  by using a list. I'll definetely do some perf testing of this, but since that's what has been shown time and time again in this code base, I'm starting with that